internal/resource/mutation/mutation.go

```diff
@@ -74,12 +77,20 @@ func (o *Op) Apply(ctx context.Context, comp *apiv1.Composition, current, mutate
 
 // unquoteKey removes quotes from a key string, handling both single and double quotes
 func unquoteKey(key string) string {
-	if matches := quotedStringRegex.FindStringSubmatch(key); matches != nil {
-		if matches[1] == matches[3] {
-			return matches[2]
-		}
+	matches := quotedStringRegex.FindStringSubmatch(key)
+	if matches == nil || matches[1] != matches[3] {
+		return key
+	}
+
+	content := matches[2]
```

Comment:
So essentially this is handling strings of the following forms:
- `"hello\"there"`
- `'hello\'there'`

right?